projects/plugins/crm/includes/ZeroBSCRM.InvoiceBuilder.php

```diff
@@ -1086,14 +1091,26 @@ function zeroBSCRM_invoicing_generateInvoiceHTML( $invoice_id = -1, $template =
 			// ignore if status_bool (non-completed status)
 			$partial['status_bool'] = (int) $partial['status_bool'];
 			if ( isset( $partial ) && $partial['status_bool'] == 1 ) { // phpcs:ignore Universal.Operators.StrictComparisons.LooseEqual
+				// v3.0+ has + or - partials. Account for that.
+				switch ( $partial['type'] ) {
+
+					case __( 'Sale', 'zero-bs-crm' ):
+						// these count as debits against invoice.
+						$balance = $balance - $partial['total'];
+						break;
+
+					case __( 'Refund', 'zero-bs-crm' ):
+					case __( 'Credit Note', 'zero-bs-crm' ):
+						// These count as credits against invoice, and should be added.
+						// If inputted as negative, then add that amount.
+						if ( str_starts_with( $partial['total'], '-' ) ) {
+							$balance = $balance + $partial['total'];
+						} else {
+							// If inputted as positive, then subtract that amount.
+							$balance = $balance - $partial['total'];
+						}
```

Comment:
And I suppose that here the same, considering the balance as the pending amount to be paid:

`$balance += abs( (float) $partial['total'] );`

Reply:
See comment above.